lib/chef/knife/base_vsphere_command.rb

```diff
@@ -114,51 +110,51 @@ def get_vim_connection
 
         vim = RbVmomi::VIM.connect conn_opts
         config[:vim] = vim
-        return vim
+        vim
       end
 
-      def get_password
-        @password ||= ui.ask("Enter your password: ") { |q| q.echo = false }
+      def password
+        @password ||= ui.ask('Enter your password: ') { |q| q.echo = false }
       end
 
       def get_vm(vmname)
-        vim = get_vim_connection
-        baseFolder = find_folder(get_config(:folder));
+        vim_connection
+        baseFolder = find_folder(get_config(:folder))
         retval = traverse_folders_for_vm(baseFolder, vmname)
-        return retval
+        retval
```

Comment:
You could reduce these last two lines even further to just `traverse_folders_for_vm(baseFolder, vmname)`